articles/javascript-higher-order-functions/index.md

```diff
@@ -0,0 +1,333 @@
+
+Functional programming is a significant building block for JavaScript as a language. They are some of the common scrips that you will continuously use when building JavaScript applications. Functions set statements to perform tasks. Functions define a task, which you can call and execute whenever you want inside your program.
+
+JavaScript treats a function as an object assigned to values to return a value. They have methods and properties, just like the other objects. Unlike ordinary objects, a function can be
+
+- Called.
+- Passed as an argument to another function.
+- Assigned to a variable as a value.
+- Be returned by another function.
+
+Functions are fundamental that outside the "normal" functions, JavaScript has high order functions.
+
+A high order function accepts another function as an argument or returns a function as its output. The concept is commonly referred to as a callback function. A function that takes another function as its argument is what we refer to as a callback function because it calls back that function.
+
+In this guide, we learn about the high order function. We will discuss and show you some of the common built-in JavaScript high order functions with examples. These include `filter`, `map`, `for each`, and `reduce`. To get along with this guide, prior knowledge to JavaScript [regular functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Functions) and [arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) will be important.
```

Comment:
Explain the 2 examples you will be considering throughout the tutorial. It is important to understand the 2 arrays we will be dealing with early on to understand the article better. Also, I feel a bit lost with the code and no outputs. It would be better if you included the outputs next to the code snippets. Detailed explanations of code are required.